x-pack/plugins/apm/server/lib/security/get_indices_privileges.ts

```diff
@@ -3,22 +3,27 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { Logger } from 'src/core/server';
 import { Setup } from '../helpers/setup_request';
 
-export async function getIndicesPrivileges(setup: Setup) {
+export async function getIndicesPrivileges(setup: Setup, logger: Logger) {
   const { client, indices } = setup;
-  const response = await client.hasPrivileges({
-    index: [
-      {
-        names: [
-          indices['apm_oss.errorIndices'],
-          indices['apm_oss.metricsIndices'],
-          indices['apm_oss.transactionIndices'],
-          indices['apm_oss.spanIndices']
-        ],
-        privileges: ['read']
-      }
-    ]
-  });
-  return response.index;
+  try {
```

Comment:
Is it possible to check if security is disabled?

Reply:
Agree. At a minimum we should check in the catch whether this was caused by a missing security endpoint (and not just swallow any error).
I'm also not sure if it's appropriate to log a warning for this. It seems a little noisy to log warnings for missing security for those who have explicitly disabled it.

Reply:
This will be possible after this PR is merged: https://github.com/elastic/kibana/pull/64651